### PR TITLE
[Tablet support M2] Full flow

### LINKED ISSUE #10568: [Tablet Support M2] Add nav host fragment to contain order detail fragment
The objective of this task is to integrate a new navigation host fragment within the order list fragment, which will encapsulate the order detail fragment. This integration will facilitate the presentation of a two-pane layout, specifically tailored for tablet devices.

### PR DESCRIPTION
**Pull request by AnirudhBhat**
Closes: #10568 

### Description

This PR introduces a `nav_host_fragment` within the `OrderListFragment`, facilitating a two-pane layout for tablets. On phones, this layout adjustment will have no impact. The `nav_host_fragment` is designated to host the Order detail screen.

**Note:** This PR solely focuses on layout adjustments and does not encompass any business logic for selecting orders or automatically opening the first order by default.

### Testing instructions

1. Run the app on physical tablet or in a tablet emulator
2. Click on the `Orders` tab
3. Ensure you see a 2 pane layout with order list and order detail
4. Go into split mode and resize the window and ensure the layout moves between 2 pane and single pane layout according to the window size
5. Run the same app on phone and ensure the behaviour is same as before and there's no change

### Images/gif

Screenshot 2024-01-23 at 9 01 26 AM

- [x] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          3 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 300 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Class <code>SelectedOrderTrackerViewModel</code> is missing tests, but <code>unit-tests-exemption</code> label was set to ignore this.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-android/milestone/195">17.5</a>. The due date for this milestone has already passed.<br />
Please assign it to a milestone with a later deadline or check whether the release for this milestone has already been finished.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false"><p>This PR contains changes to Tracks-related logic. Please ensure (<strong>author and reviewer</strong>) the following are completed:</p>

<ul>
  <li>The tracks events must be validated in the Tracks system.</li>
  <li>Verify the internal Tracks spreadsheet has also been updated.</li>
  <li>Please consider registering any new events.</li>
  <li>The PR must be assigned the <strong>category: tracks</strong> label.</li>
</ul></td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/src/main/res/layout/fragment_order_list.xml [hunk 1]
```diff
@@ -1,81 +1,110 @@
...
+                android:layout_margin="@dimen/major_100"
+                android:contentDescription="@string/orderlist_create_order_button_description"
+                app:layout_constraintBottom_toBottomOf="parent"
+                app:layout_constraintEnd_toEndOf="parent" />
+
+        </androidx.constraintlayout.widget.ConstraintLayout>
+
+    </com.woocommerce.android.widgets.ScrollChildSwipeRefreshLayout>
+
+    <LinearLayout
```
Comment (samiuelson) [inline WooCommerce/src/main/res/layout/fragment_order_list.xml]: ❓ Why do we need to wrap <fragment> with LinearLayout?

**Review by samiuelson — COMMENTED** [review]
Looks good overall. I wonder if we could simplify the XML layout slightly. Let me know what you think.

File: WooCommerce/src/main/res/layout/fragment_order_list.xml (hunk 1 above)
Comment (AnirudhBhat) [reply WooCommerce/src/main/res/layout/fragment_order_list.xml]: Good point. Removed unnecessary LinearLayout here: https://github.com/woocommerce/woocommerce-android/pull/10591/commits/ee6de645c029fedff0c59d1f002f81cfa4635e1d

**Review by samiuelson — APPROVED** [review]
LGTM

**Comment by AnirudhBhat** [comment]
@pachlava The UI test failure in this PR appears to be related to the `constructor() : super(R.id.toolbar)` in `FilterScreen`. This issue likely stems from our refactoring, where we transitioned to using fragment-specific toolbars in place of a shared toolbar.
cc @samiuelson

**Comment by AnirudhBhat** [comment]
@pachlava The UI test failure in this PR appears to be related to the `constructor() : super(R.id.toolbar)` in `FilterScreen`. This issue likely stems from our recent refactoring, where we transitioned to using fragment-specific toolbars in place of a shared toolbar.

File: RELEASE-NOTES.txt [hunk 2]
```diff
@@ -2,6 +2,11 @@
 
 17.4
 -----
+- [***] The Order Listing and Order Detail screens have been optimized for tablet use, showcasing a two-pane layout on such devices. [https://github.com/woocommerce/woocommerce-android/pull/10591]
+
+17.3
+-----
+
```
Comment (samiuelson) [inline RELEASE-NOTES.txt]:
```suggestion
```
💡 Duplicate

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/main/MainActivity.kt [hunk 3]
```diff
@@ -899,10 +929,34 @@ class MainActivity :
         }
     }
 
-    private fun navigateToWebView(event: ViewUrlInWebView) {
+    fun navigateToWebView(event: ViewUrlInWebView) {
```
Comment (samiuelson) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/main/MainActivity.kt]:
```suggestion
    private fun navigateToWebView(event: ViewUrlInWebView) {
```
💡 This should be private

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/main/MainActivityViewModel.kt [hunk 4]
```diff
@@ -398,7 +398,11 @@ class MainActivityViewModel @Inject constructor(
     object ViewPayments : Event()
     object ViewTapToPay : Event()
     object RequestNotificationsPermission : Event()
-    data class ViewUrlInWebView(val url: String) : Event()
+    data class ViewUrlInWebView(
+        val url: String,
+        val urlsToTriggerExit: Array<String>? = null,
+        val title: String? = null,
```
Comment (samiuelson) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/main/MainActivityViewModel.kt]:
```suggestion
```
❓ These props are not used anywhere

File: RELEASE-NOTES.txt (hunk 2 above)
Comment (AnirudhBhat) [reply RELEASE-NOTES.txt]: Correct. Fixed here: https://github.com/woocommerce/woocommerce-android/pull/10591/commits/5aab6a91ce70aebafa8697469923774268db1ba7

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/main/MainActivity.kt (hunk 3 above)
Comment (AnirudhBhat) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/main/MainActivity.kt]: Good catch: Done here: https://github.com/woocommerce/woocommerce-android/pull/10591/commits/b84663b290bba80a37f959f9ef0fb92c855f23cf

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/main/MainActivityViewModel.kt (hunk 4 above)
Comment (AnirudhBhat) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/main/MainActivityViewModel.kt]: Correct. Removed it here: https://github.com/woocommerce/woocommerce-android/pull/10591/commits/967ad7e9a3678e8075d5ee7b1b599f712402bb4d

**Comment by samiuelson** [comment]
@AnirudhBhat, as shared in the test charter with more details (pdfdoF-4C0-p2#comment-5822), there are the following navigation scenarios that I believe should be fixed before release:

##### Issue 1
1. Open the app, go to Orders, and wait for the first order to load on the right pane
2. Click back – you'll be redirected to My store
3. Go to Orders again
Observe order is not loading – the pane on the right side displays shimmer animation endlessly.

##### Issue 2
1. Open the app in split mode, go to Orders, and wait for the first order to load on the right pane
2. Adjust Woo app's window size, so that it displays single pane only, the order details screen shuold become visible
3. Click arrow back on the toolbar
Observe there is enless shimmer animation displayed on the screen instead of the orders list screen. We are not able to get back to orders list nor by clicking back arrow navigation or system back press.

##### Issue 3
1. Open the app, go to Orders, and wait for the first order to load on the right pane
2. Click search icon and enter search phrase, e.g. existing order number
3. Click on the search result to load it in the right-side pane
Observe that the order list shows only search results (not the full list of orders) and there is no way to go back to the full orders list. Clicking "back" redirects to My Store instead of going back to full list.

**Comment by AnirudhBhat** [comment]
@samiuelson Thanks for the review and testing! Really appreciate it!

I have fixed issue 1 and 3 here: https://github.com/woocommerce/woocommerce-android/pull/10591/commits/f94ef0aaa2502f7b6094bb15cac310a5228d1e50 https://github.com/woocommerce/woocommerce-android/pull/10591/commits/7d16c9903a127cf3e580c9401fb54164d7a968d1

I need additional time to resolve issue 2. Given that it's an edge case, occurring only during the transition from tablet to non-tablet mode, I suggest proceeding with the release. The impact on users is minimal, as navigating to a different tab and returning to the orders tab serves as a workaround. What are your thoughts?

**Review by samiuelson — APPROVED** [review]
Good work, @AnirudhBhat! LGTM. 

It looks like the UI tests (e2eRealApiOrdersFilter) should still be updated to make the checks pass.

Please also make sure to add any pending issues to the project board.

**Comment by samiuelson** [comment]
Adding the "do not merge" label until we make a final decision on whether to include it in this week's release or not. cc @malinajirka

**Comment by AnirudhBhat** [comment]
@pachlava @samiuelson Heads up. I have added `@Ignore` annotation for couple of UI tests that need some fix due to our recent toolbar refactoring. https://github.com/woocommerce/woocommerce-android/pull/10591/commits/b85d5625f592574fadb03273f2e5feb5ba9f510a

**Comment by malinajirka** [comment]
Thank you both! We have found some issues today, which we fixed. However, it seems having an additional week for testing will increase our confidence. Moreover, I don't think we should merge a PR with broken tests - that beats their purpose. Let's move the target date to the next week.

File: WooCommerce/src/androidTest/kotlin/com/woocommerce/android/e2e/tests/ui/OrdersUITest.kt [hunk 5]
```diff
@@ -48,6 +49,7 @@ class OrdersUITest : TestBase() {
     }
 
     @Test
+    @Ignore
```
Comment (kidinov) [inline WooCommerce/src/androidTest/kotlin/com/woocommerce/android/e2e/tests/ui/OrdersUITest.kt:52]: Just noticed that 2 tests are ignored now

**Pull request merged**